src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearing/WithdrawHearingAction.java

```diff
@@ -0,0 +1,189 @@
+
+public class WithdrawHearingAction extends Action implements RulesProcessable {
+
+    protected WithdrawHearingRequest withdrawHearingRequest;
+    protected Hearing hearing;
+    protected List<HearingPart> hearingParts;
+    protected List<Session> sessions;
+
+    protected HearingRepository hearingRepository;
+    protected HearingPartRepository hearingPartRepository;
+    protected StatusConfigService statusConfigService;
+    protected StatusServiceManager statusServiceManager;
+
+    // id & hearing part string
+    private Map<UUID, String> originalHearingParts;
+    private String previousHearing;
+
+    public WithdrawHearingAction(
+        WithdrawHearingRequest withdrawHearingRequest,
+        HearingRepository hearingRepository,
+        HearingPartRepository hearingPartRepository,
+        StatusConfigService statusConfigService,
+        StatusServiceManager statusServiceManager,
+        ObjectMapper objectMapper
+    ) {
+        this.withdrawHearingRequest = withdrawHearingRequest;
+        this.hearingRepository = hearingRepository;
+        this.hearingPartRepository = hearingPartRepository;
+        this.statusConfigService = statusConfigService;
+        this.statusServiceManager = statusServiceManager;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public void getAndValidateEntities() {
+        hearing = hearingRepository.findOne(withdrawHearingRequest.getHearingId());
+        hearingParts = hearing.getHearingParts();
+        sessions = hearingParts.stream()
+            .map(HearingPart::getSession)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+        // Validation moved to act() due to conflict with optimistic lock
+
+        if (!statusServiceManager.canBeWithdrawn(hearing)) {
+            throw new SnlEventsException("Hearing can not be withdrawn");
+        }
+        hearingParts.forEach(hp -> {
+            if (!statusServiceManager.canBeWithdrawn(hp)) {
+                // we should define somewhere text of these messages and how much we want to show to the user
+                throw new SnlEventsException("Hearing part can not be withdrawn");
+            }
+        });
+    }
+
+    @Override
+    public UUID[] getAssociatedEntitiesIds() {
+        val ids = hearing.getHearingParts().stream().map(HearingPart::getId).collect(Collectors.toList());
+        ids.addAll(hearing.getHearingParts().stream()
+            .map(HearingPart::getSessionId)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList()));
+        ids.add(withdrawHearingRequest.getHearingId());
+
+        return ids.stream().toArray(UUID[]::new);
+    }
+
+    @Override
+    public void act() {
+        hearing.setStatus(statusConfigService.getStatusConfig(Status.Withdrawn));
+
+        originalHearingParts = mapHearingPartsToStrings(hearingParts);
+        hearingParts.stream().forEach(hp -> {
+            VersionInfo vi = getVersionInfo(hp);
```

Comment:
You need to remove the link between hearingPart and session here. So set sessionId to null and Session to null.

Reply:
Done